File: expo-ai-composer/ios/ExpoAiComposerModule.swift
import ExpoModulesCore

public class ExpoAiComposerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoAiComposer")

    Constants([
      "defaultMinHeight": 48.0,
      "defaultMaxHeight": 120.0,
      "contentGap": 0.0
    ])

    // Composer View definition
    View(AiComposerView.self) {
      Prop("placeholder") { (view: AiComposerView, value: String) in
        view.placeholder = value
      }

      Prop("text") { (view: AiComposerView, value: String) in
        view.text = value
      }

      Prop("minHeight") { (view: AiComposerView, value: CGFloat) in
        view.minHeight = value
      }

      Prop("maxHeight") { (view: AiComposerView, value: CGFloat) in
        view.maxHeight = value
      }

      Prop("sendButtonEnabled") { (view: AiComposerView, value: Bool) in
        view.sendButtonEnabled = value
      }

      Prop("showSendButton") { (view: AiComposerView, value: Bool) in
        view.showSendButton = value
      }

      Prop("editable") { (view: AiComposerView, value: Bool) in
        view.editable = value
      }

      Prop("autoFocus") { (view: AiComposerView, value: Bool) in
        view.autoFocus = value
      }

      Prop("isStreaming") { (view: AiComposerView, value: Bool) in
        view.isStreaming = value
      }

      Prop("expandedEditorEnabled") { (view: AiComposerView, value: Bool) in
        view.expandedEditorEnabled = value
      }

      // Imperative view functions (dispatched on the main queue by the view DSL).
      AsyncFunction("focus") { (view: AiComposerView) in
        view.focus()
      }

      AsyncFunction("blur") { (view: AiComposerView) in
        view.blur()
      }

      AsyncFunction("clear") { (view: AiComposerView) in
        view.clear()
      }

      Events(
        "onChangeText",
        "onSend",
        "onStop",
        "onHeightChange",
        "onKeyboardHeightChange",
        "onComposerFocus",
        "onComposerBlur"
      )
    }

    // Keyboard-aware wrapper view
    View(AiComposerWrapper.self) {
      Prop("pinToTopEnabled") { (view: AiComposerWrapper, value: Bool) in
        view.pinToTopEnabled = value
      }

      Prop("extraBottomInset") { (view: AiComposerWrapper, value: CGFloat) in
        view.extraBottomInset = value
      }

      // No-op on iOS (kept for cross-platform API parity)
      Prop("extraTopInset") { (_: AiComposerWrapper, _: CGFloat) in }

      Prop("scrollToTopTrigger") { (view: AiComposerWrapper, value: Double) in
        view.scrollToTopTrigger = value
      }
    }
  }
}

File: expo-ai-composer/android/src/main/java/expo/modules/expoaicomposer/ExpoAiComposerModule.kt
package expo.modules.expoaicomposer

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Expo module for AiComposer.
 */
class ExpoAiComposerModule : Module() {

    override fun definition() = ModuleDefinition {
        Name("ExpoAiComposer")

        // AiComposerView
        View(AiComposerView::class) {
            Prop("placeholder") { view: AiComposerView, value: String ->
                view.placeholderText = value
            }

            Prop("text") { view: AiComposerView, value: String ->
                view.text = value
            }

            Prop("minHeight") { view: AiComposerView, value: Float ->
                view.minHeightDp = value
            }

            Prop("maxHeight") { view: AiComposerView, value: Float ->
                view.maxHeightDp = value
            }

            Prop("sendButtonEnabled") { view: AiComposerView, value: Boolean ->
                view.sendButtonEnabled = value
            }

            Prop("showSendButton") { view: AiComposerView, value: Boolean ->
                view.showSendButton = value
            }

            Prop("editable") { view: AiComposerView, value: Boolean ->
                view.editable = value
            }

            Prop("autoFocus") { view: AiComposerView, value: Boolean ->
                view.autoFocus = value
            }

            Prop("isStreaming") { view: AiComposerView, value: Boolean ->
                view.isStreaming = value
            }

            // Imperative view functions (dispatched on the main queue by the view DSL).
            AsyncFunction("focus") { view: AiComposerView ->
                view.focus()
            }

            AsyncFunction("blur") { view: AiComposerView ->
                view.blur()
            }

            AsyncFunction("clear") { view: AiComposerView ->
                view.clear()
            }

            Events(
                "onChangeText",
                "onSend",
                "onStop",
                "onHeightChange",
                "onKeyboardHeightChange",
                "onComposerFocus",
                "onComposerBlur"
            )
        }

        // AiComposerWrapper
        View(AiComposerWrapper::class) {
            Prop("pinToTopEnabled") { view: AiComposerWrapper, value: Boolean ->
                view.pinToTopEnabled = value
            }

            Prop("extraBottomInset") { view: AiComposerWrapper, value: Float ->
                view.extraBottomInset = value
            }

            Prop("scrollToTopTrigger") { view: AiComposerWrapper, value: Double ->
                view.scrollToTopTrigger = value
            }
        }
    }
}
